src/idfxt/JSONDataInterface.h

```diff
@@ -0,0 +1,102 @@
+// copyright 2014, Shannon Mackey <mackey@BUILDlab.net>
+#ifndef JSONDATAINTERFACE_H
+#define JSONDATAINTERFACE_H
+
+//for JSONDataObject and JSONDataInterface
+#include <string>
+#include <list>
+#include <map>
+#include <memory>
+
+//for old IDF text import only
+#include <algorithm>
+#include <cctype>
+#include <cstring>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <vector>
+#include <chrono>
+#include <random>
+
+const std::string ACCEPTED_VERSION = "8.2";
+//end old IDF text import only
+
+extern "C"
+{
+#include "../../third_party/cJSON/cJSON.h"
+}
+
+namespace idfx {
+
+
+class JSONDataObject
+{
+public:
+    JSONDataObject(const std::string &json_content);
+    ~JSONDataObject();
+
+    std::map<std::string, std::string> getProperties();
+    std::string getPropertyValue(std::string);
+//    void setPropertyValue(std::string, std::string);
+//    void setPropertyValue(std::string, int);
+//    void setPropertyValue(std::string, double);
+
+    std::list<std::shared_ptr<JSONDataObject> > getExtensions();
+
+    
+    std::string print();
+    //friend access private constructor to avoid conversion to/from strings internally
+//    friend std::list<JSONDataObject> JSONDataInterface::getModelObjects(std::string);
+
+
+private:
+    cJSON *data_j;
+
+    void c_getChildren(std::list< std::shared_ptr< idfx::JSONDataObject > > &return_object_list, cJSON *subitem);
+//    JSONDataObject(cJSON &cjson); //to hide cJSON pointer
+};
+
+
+//////////////////////////////////////////////////////////
+
+class JSONDataInterface
+{
+public:
+    JSONDataInterface(const std::string &json_schema);
+    ~JSONDataInterface();
+
+    std::list<std::shared_ptr<JSONDataObject> > getModelObjects(std::string object_type = "");
```

Comment:
Empty string is to return all objects.  The object_type parameter was a
suggestion of possible convenience method, but remains un-implemented.  In
light of the discussion previously, on the topics of convenience methods -
I'm leaving this un-implemented until such time that I need it in the
course of getting E+ to accept the JSON files.

On Fri, Jan 16, 2015 at 6:44 PM, Dan Macumber <notifications@github.com>
wrote:

> In src/idfxt/JSONDataInterface.h
> <https://github.com/NREL/EnergyPlus/pull/4636#discussion-diff-23117736>:
>
> > +    cJSON *data_j;
> > +
> > +    void c_getChildren(std::list< std::shared_ptr< idfx::JSONDataObject > > &return_object_list, cJSON *subitem);
> > +//    JSONDataObject(cJSON &cjson); //to hide cJSON pointer
> > +};
> > +
> > +
> > +//////////////////////////////////////////////////////////
> > +
> > +class JSONDataInterface
> > +{
> > +public:
> > +    JSONDataInterface(const std::string &json_schema);
> > +    ~JSONDataInterface();
> > +
> > +    std::list<std::shared_ptr<JSONDataObject> > getModelObjects(std::string object_type = "");
>
> @refaqtor <https://github.com/refaqtor> does empty string return all
> objects or nothing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NREL/EnergyPlus/pull/4636/files#r23117736>.
>